components/eam/src/physics/cam/cldera_dynamic_tracers.F90

```diff
@@ -0,0 +1,371 @@
+  public :: cldera_dynamic_tracers_pv_idx          ! getter function for PV tracer global constituent index
+  public :: cldera_dynamic_tracers_pt_idx          ! getter function for PT tracer global constituent index
+  public :: cldera_dynamic_tracers_is_pt           ! return true if named tracer is PT tracer
+  public :: cldera_dynamic_tracers_is_pv           ! return true if names tracer is PV tracer
+  public :: cldera_dynamic_tracers_pv_offset       
+
+  real(r8), parameter :: cldera_dynamic_tracers_pv_offset = 1_r8
+
+  ! ----- Private module data
+  integer, parameter :: ncnst=2  ! number of constituents implemented by this module
+
+  ! constituent names
+  character(len=8), parameter :: c_names(ncnst) = (/'PV_TRCR_   ', 'PT_TRCR    '/)
+  character(len=8), parameter :: pv_out_name = 'PV_TRCR    '
+
+  integer :: ifirst ! global index of first constituent
+  integer :: ixpv   ! global index for PV tracer
+  integer :: ixpt   ! global index for PT tracer
+
+  ! Data from namelist variables
+  logical :: cldera_dynamic_tracers_flag  = .false. ! true => activate this module, namelist variable
+
+!===============================================================================
+contains
+!===============================================================================
+
+  subroutine cldera_dynamic_tracers_readnl(nlfile)
+
+    use namelist_utils, only: find_group_name
+    use units,          only: getunit, freeunit
+    use mpishorthand
+    use cam_abortutils, only: endrun
+
+    implicit none
+
+    character(len=*), intent(in) :: nlfile  ! filepath for file containing namelist input
+
+    ! Local variables
+    integer :: unitn, ierr
+    character(len=*), parameter :: subname = 'cldera_dynamic_tracers_readnl'
+
+
+    namelist /cldera_dynamic_tracers_nl/ cldera_dynamic_tracers_flag
+
+    !-----------------------------------------------------------------------------
+
+    if (masterproc) then
+       unitn = getunit()
+       open( unitn, file=trim(nlfile), status='old' )
+       call find_group_name(unitn, 'cldera_dynamic_tracers_nl', status=ierr)
+       if (ierr == 0) then
+          read(unitn, cldera_dynamic_tracers_nl, iostat=ierr)
+          if (ierr /= 0) then
+             call endrun(subname // ':: ERROR reading namelist')
+          end if
+       end if
+       close(unitn)
+       call freeunit(unitn)
+    end if
+
+#ifdef SPMD
+    call mpibcast(cldera_dynamic_tracers_flag, 1, mpilog,  0, mpicom)
+#endif
+
+  endsubroutine cldera_dynamic_tracers_readnl
+
+!================================================================================
+
+  subroutine cldera_dynamic_tracers_register
+    !----------------------------------------------------------------------- 
+    ! 
+    ! Purpose: register advected constituents
+    ! 
+    !-----------------------------------------------------------------------
+    use physconst,  only: cpair, mwdry
```

Comment:
It does not look like these constants are used in this routine, please remove.

Reply:
removed